build.zig

```diff
@@ -359,9 +365,109 @@ fn build_tigerbeetle(
         target: std.Build.ResolvedTarget,
         mode: std.builtin.OptimizeMode,
         tracer_backend: config.TracerBackend,
+        multiversion: ?[]const u8,
         emit_llvm_ir: bool,
     },
 ) void {
+    const out_filename = if (options.target.result.os.tag == .windows)
+        "tigerbeetle.exe"
+    else
+        "tigerbeetle";
+
+    const tigerbeetle_bin = if (options.multiversion) |version| bin: {
+        assert(!options.emit_llvm_ir);
+        const build_mutliversion_exe = b.addExecutable(.{
+            .name = "build_mutliversion",
```

Comment:
:heart: 

I would like in general to brainstorm some better name for the concept! `multiversion`  is somewhat hard to mentally parse for me, is long, and we use both `multiversion` and `multiversioning`...

Reply:
It's an oddity too, because I actually settled on calling everything else `release` internally. So `multirelease` would technically make more sense for the concept name, but it felt even worse...